ui/src/contexts/Identity.tsx

```diff
@@ -0,0 +1,157 @@
+import { message } from "antd";
+import {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { useLocation, useNavigate, useSearchParams } from "react-router-dom";
+
+import { getIdentities } from "src/adapters/api/identities";
+import { useEnvContext } from "src/contexts/Env";
+import { AppError, Identity } from "src/domain";
+import { ROUTES } from "src/routes";
+import { AsyncTask, isAsyncTaskDataAvailable } from "src/utils/async";
+import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
+import { IDENTIFIER_SEARCH_PARAM, ROOT_PATH } from "src/utils/constants";
+import { buildAppError } from "src/utils/error";
+
+type IdentityState = {
+  fetchIdentities: (signal: AbortSignal) => void;
+  handleChange: (identifier: string) => void;
+  identifier: string;
+  identitiesList: AsyncTask<Identity[], AppError>;
+  identityDisplayName: string;
+};
+
+const defaultIdentityState: IdentityState = {
+  fetchIdentities: () => void {},
+  handleChange: () => void {},
+  identifier: "",
+  identitiesList: { status: "pending" },
+  identityDisplayName: "",
+};
+
+const IdentityContext = createContext(defaultIdentityState);
+
+export function IdentityProvider(props: PropsWithChildren) {
+  const env = useEnvContext();
+  const [messageAPI, messageContext] = message.useMessage();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [identitiesList, setIdentitiesList] = useState<AsyncTask<Identity[], AppError>>({
+    status: "pending",
+  });
+  const [identifier, setIdentifier] = useState("");
```

Comment:
Why do we need to declare this state if we are storing the selected identifier in the URL?

Reply:
because when we navigate to some page we lose the identifier in the URL and set it each time from a state